src/components/tooltip/tooltip.ts

```diff
@@ -142,6 +144,23 @@ export class MdTooltip {
    * Hides the tooltip and returns a promise that will resolve when the tooltip is hidden
    */
   hide(): Promise<any> {
+    if (this._tooltipRef) {
+      return new Promise(done => {
+        this._tooltipRef.remove().then(() => {
+          this.destroy().then(() => done());
+        });
+        this._tooltipRef = null;
+      });
```

Comment:
I think this block could just be:
```ts
return this._tooltipRef.remove()
    .then(() => this.destroy())
  this._tooltipRef = null;  
```

Setting the tooltipRef to null doesn't need to be part of the chain, and you already have a promise coming back from both `remove` and `destroy`, so you shouldn't need a new one.